docs/cpp_client/cpp_examples.md

```diff
@@ -45,20 +48,25 @@ camera_config.from_json(file);
 Alternatively, the `struct` can be setup directly in the code:
 
 ```cpp
-CameraConfig camera_config;
-camera_config.server_ip = ip;
-camera_config.listen_port = 8100;
-camera_config.location.z = 200;
-camera_config.rotation.pitch = -5;
-camera_config.resolution = CameraConfig::Resolution(1024,1024);
+// define desired sensors
+    std::vector<std::shared_ptr<Sensor>> sensors;
+
+    CameraConfig fc_config;
+    fc_config.server_ip = sim0.getServerIp();
+    fc_config.server_port = sim0.getServerPort();
+    fc_config.listen_port = 8100;
+    fc_config.location.z = 225;
+    fc_config.rotation.pitch = -5;
+    fc_config.resolution = Resolution(1920,1080);
+    fc_config.annotation.include_annotation = true;
+    fc_config.annotation.desired_tags = {"traffic_sign"};
 ```
 
 After the sensor's configuration is created, all that needs to be done is 
 configure the sensor:
 
 ```cpp
-Sensor camera(CameraConfig);
-camera.configure();
+sensors.push_back(std::make_shared<Sensor>(std::make_unique<CameraConfig>(fc_config)));
```

Comment:
This doesn't actually configure the sensor. This just adds it to the list of sensor to be configured. The configuration happens here: https://github.com/monoDriveIO/monodrive-client/blob/master/examples/cpp/dev/camera_sensor.cpp#L73

Reply:
So, do we want to include this part in this example? 
```
Alternatively, the `struct` can be setup directly in the code:

```cpp
// define desired sensors
    std::vector<std::shared_ptr<Sensor>> sensors;

    CameraConfig fc_config;
    fc_config.server_ip = sim0.getServerIp();
    fc_config.server_port = sim0.getServerPort();
    fc_config.listen_port = 8100;
    fc_config.location.z = 225;
    fc_config.rotation.pitch = -5;
    fc_config.resolution = Resolution(1920,1080);
    fc_config.annotation.include_annotation = true;
    fc_config.annotation.desired_tags = {"traffic_sign"};
```

After the sensor's configuration is created, all that needs to be done is 
configure the sensor:

```cpp
sensors.push_back(std::make_shared<Sensor>(std::make_unique<CameraConfig>(fc_config)));
for (auto& sensor : sensors)
{
    sensor->configure();
}
``````

Reply:
`sensors` is an array that all the configuration are stored in. The call in the loop is what actually configures the sensor. So both parts are needed to fully configure everything.

If you take a look at the viewport sensor, this slightly different because we don't store it in the `sensors` array. It is configured directly after casting it to a shared pointer.

Reply:
Great, learning a little bit of C++ too.